### PR TITLE
Add providers

### PR DESCRIPTION
**Pull request by yann300**
This PR automatically add hardhat, ganache and wallet connect provider when Remix loads.
see https://github.com/bunsenstraat/remix-walletconnect/pull/1 for wallect connect compatibility

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/abstract-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import * as packageJson from '../../../../../package.json'
+import { Plugin } from '@remixproject/engine'
+import { AppModal, AlertModal, ModalTypes } from '@remix-ui/app'
+import React from 'react' // eslint-disable-line
+import { Blockchain } from '../../blockchain/blockchain'
+import { ethers } from 'ethers'
+
+const profile = {
+  name: 'hardhat-provider',
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/tabs/abstract-provider.tsx:9]: This name should be updated

**Review by Aniket-Engg — CHANGES_REQUESTED** [review]

File: libs/remix-ui/run-tab/src/lib/actions/index.ts [hunk 2]
```diff
@@ -223,8 +197,8 @@ export const setNetworkNameFromProvider = (networkName: string) => {
 }
 
 const addExternalProvider = (network) => {
+  console.log('adding ext provider', network)
```
Comment (ioedeveloper) [inline libs/remix-ui/run-tab/src/lib/actions/index.ts:200]: Do we need this log?

**Review by ioedeveloper — DISMISSED** [review]
Ganache and Hardhat provider modal only pops up once. If the cancel button is clicked to close the modal, the modal no longer pops up.

**Comment by yann300** [comment]
> Ganache and Hardhat provider modal only pops up once. If the cancel button is clicked to close the modal, the modal no longer pops up.

done

File: package.json [hunk 3]
```diff
@@ -98,6 +98,7 @@
     "nightwatch_local_migrate_filesystem": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/migrateFileSystem.test.js --env=chrome",
     "nightwatch_local_stress_editor": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/stressEditor.test.js --env=chromeDesktop",
     "nightwatch_local_search": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/search.test.js --env=chromeDesktop",
+    "nightwatch_local_search": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/providers.test.js --env=chromeDesktop",
```
Comment (Aniket-Engg) [inline package.json:101]: script name should be fixed

Comment (yann300) [reply package.json]: done

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**